service/src/main/java/bio/terra/workspace/service/petserviceaccount/PetSaService.java

```diff
@@ -232,6 +233,27 @@ private static boolean bindingEquals(Binding binding1, Binding binding2) {
     return serviceAccountExists(constructedSa) ? Optional.of(constructedSa) : Optional.empty();
   }
 
+  /**
+   * Fetches credentials for the provided user's pet service account in the current workspace's GCP
+   * context if one exists. This will create a pet SA for the user if one does not exist, but will
+   * return empty if the workspace does not have a GCP context.
+   *
+   * <p>This method does not validate that the provided credentials have appropriate workspace
+   * access.
+   */
+  public Optional<AuthenticatedUserRequest> getWorkspacePetCredentials(
+      UUID workspaceId, AuthenticatedUserRequest userRequest) {
```

Comment:
minor: since you have the userRequest, you could make this method always return an AuthenticatedUserRequest - either the pet or the userRequest.

But maybe you need this method anyway, so could make a `getPetCredsIfTheyExist()` I suppose.
Anyway, this is mostly aesthetic, so feel free to ignore.

Reply:
It's a  bit of an awkward contract for a public method, because the SA creds don't belong to the workspace but instead to a cloud context's GCP project. However, if you just returned an `AuthenticatedUserRequest`, you'd have no way of knowing whether it was the pet or not without inspecting the email address. Maybe there's an opportunity to make that class more expressive.

I'd keep it the way it is for now and start thinking about higher-level ways of working with different kinds of credentials.